### ISSUE TITLE
Document Viewer: Upgrade PDF.js 2.9.35

### ISSUE THREAD

**Issue by melloware** [primefaces-extensions/primefaces-extensions#122]
**Is your feature request related to a problem? Please describe.**
It has been a few years since the underlying PDF.js lib has been updated under Document Viewer.  There was an issue with fonts in a IFRAME but it was resolved recently.

https://github.com/mozilla/pdf.js/issues/8271

**Comment by Cantfirmed** [comment]
When is the next update planned? I would do it myself but the changes.txt for documentviewer is outdated.

**Comment by melloware** [comment]
When I tried last time to upgrade the issue i mentioned above was still happening causing the font to be messed up.  I know they did a release recently but I haven't tried to upgrade it.

Is there some feature you need that is not in the current PDF viewer?

**Comment by Cantfirmed** [comment]
Some browsers (eg. Firefox, IE) have problems with visible digital signatures. They get removed in the online viewer and they are also removed if you download them. I tested it with the newest pdf.js where it now works to display the signatures correctly.

**Comment by melloware** [comment]
OK let me know because they previously had not supported digital signatures.

**Comment by Cantfirmed** [comment]
They now show them but the signatures are not validated.
https://github.com/mozilla/pdf.js/pull/13214

**Comment by melloware** [comment]
Looks like that feature its not in an official release yet as latest released was 2.8.335 on Apr 5 and this was merged on April 10th.

**Comment by melloware** [comment]
And you are right the structure of PDF.JS has changed significantly so our old upgrade instructions are completely irrelevant

**Comment by Cantfirmed** [comment]
> Looks like that feature its not in an official release yet as latest released was 2.8.335 on Apr 5 and this was merged on April 10th.

Ah, true that's my bad. I just cloned the repo and forgot to check the actual releases. The feature is already in the pre-release but I guess you want to wait for the actual release?

**Comment by melloware** [comment]
I spent about an hour trying to upgrade to v2.9.359 and it is fighting me.  Feel free to take a crack at it...i will gladly accept a PR.

**Comment by ivy-cst** [comment]
We have the same bug as here: https://github.com/wojtekmaj/react-pdf/issues/819.
With Chrome 92 and up some pdf's are not rendered correctly with the current included pdf.js.
The upgrade to pdf.js v2.9.359 would fix our problems.

**Comment by melloware** [comment]
@ivy-cst thanks for reporting.  Yuck!   Upgrading this library is quite difficult and since we use it in an IFRAME the last time I tried to upgrade it the Font would not appear.  I will put this on my list to try and upgrade.

**Comment by melloware** [comment]
OK man this was a lot of hacking but I think I got everything working with the latest PDF.js: 2.9.35.

I published a 10.0.5-SNAPSHOT of PFE if you want to give it a try...

You will need this in your pom.xml to reach the Sonatype Snapshots...
```xml
<repository>
    <id>sonatype-snapshots</id>
    <name>Sonatype Snapshot Repository</name>
    <url>https://s01.oss.sonatype.org/content/repositories/snapshots</url>
    <releases>
        <enabled>false</enabled>
    </releases>
    <snapshots>
        <enabled>true</enabled>
    </snapshots>
</repository>
```

**Comment by ivy-cst** [comment]
> OK man this was a lot of hacking but I think I got everything working with the latest PDF.js: 2.9.35.
> 
> I published a 10.0.5-SNAPSHOT of PFE if you want to give it a try...
> 
> You will need this in your pom.xml to reach the Sonatype Snapshots...
> 
> ```
> <repository>
>     <id>sonatype-snapshots</id>
>     <name>Sonatype Snapshot Repository</name>
>     <url>https://s01.oss.sonatype.org/content/repositories/snapshots</url>
>     <releases>
>         <enabled>false</enabled>
>     </releases>
>     <snapshots>
>         <enabled>true</enabled>
>     </snapshots>
> </repository>
> ```

Thank you very much. Great job. Works now with newest browsers. Would it be possible to patch that down to 7.0.x and 8.0.x? I patched 7.0.3 (the version we currently use) by hand and it did also work.

**Comment by melloware** [comment]
We are a small project here so I typically don't backport fixes to older versions and publish them to central.  We typically just always work on the latest version but as you saw patching it yourself is not too bad.

**Comment by melloware** [comment]
Anyone watching this ticket this is now in Maven Central as 10.0.5

**Comment by melloware** [comment]
I also just backported this and released PFE 8.0.5
